`lib/events/filesessions` test setup flakiness

## Failure

#### Link(s) to logs

https://github.com/gravitational/teleport/actions/runs/21212336907/job/61023905607#step:6:510

#### Relevant snippet

```
CGO_ENABLED=1 CGO_LDFLAGS="-Wl,-Bstatic  -Wl,-Bdynamic -Wl,--as-needed" go test -json -tags "pam  libfido2  pivtest "  github.com/gravitational/teleport/tool/tsh/... -race -shuffle on  \
	| gotestsum --junitfile /__w/teleport/teleport/test-logs/unit-tests-tsh.xml --jsonfile /__w/teleport/teleport/test-logs/unit-tests-tsh.json --raw-command -- cat
CGO_ENABLED=1 CGO_CFLAGS="-I/usr/libbpf-1.2.2/include" CGO_LDFLAGS="-Wl,-Bstatic -L/usr/libbpf-1.2.2/lib64 -lbpf -lelf -lz -Wl,-Bdynamic -Wl,--as-needed" go test -json -tags "pam  bpf" -test.run=TestChaos ./lib/events/filesessions \
	| gotestsum --junitfile /__w/teleport/teleport/test-logs/unit-tests-chaos.xml --jsonfile /__w/teleport/teleport/test-logs/unit-tests-chaos.json --raw-command -- cat
fatal: No names found, cannot describe anything.
GOWORK=off CGO_ENABLED=0 go -C build.assets/tooling run ./cmd/apiversion "18.6.4" > api/version.go
    Updating git repository `[https://github.com/gravitational/boring`](https://github.com/gravitational/boring%60)
✖  lib/events/filesessions

=== Failed
=== FAIL: lib/events/filesessions  (0.00s)
FAIL	github.com/gravitational/teleport/lib/events/filesessions [setup failed]

DONE 0 tests, 1 failure in 0.131s
```

I prepared a draft PR to debug this by skipping `gotestsum` to reveal the underlying test output. I managed to reproduce the build failure, and it looks like it boils down to a failure to properly autogenerate the API version info. Here's an excerpt from a [successful pass](https://github.com/gravitational/teleport/actions/runs/21438664959/job/61743653941?pr=63235#step:6:505):

```
CGO_ENABLED=1 CGO_CFLAGS="-I/usr/libbpf-1.2.2/include" CGO_LDFLAGS="-Wl,-Bstatic -L/usr/libbpf-1.2.2/lib64 -lbpf -lelf -lz -Wl,-Bdynamic -Wl,--as-needed" go test -json -tags "pam  bpf" -test.run=TestChaos ./lib/events/filesessions
make[1]: Leaving directory '/__w/teleport/teleport'
```

And here's how the [failure](https://github.com/gravitational/teleport/actions/runs/21438664959/job/61746686669?pr=63235#step:6:503) looks like:

```
CGO_ENABLED=1 CGO_CFLAGS="-I/usr/libbpf-1.2.2/include" CGO_LDFLAGS="-Wl,-Bstatic -L/usr/libbpf-1.2.2/lib64 -lbpf -lelf -lz -Wl,-Bdynamic -Wl,--as-needed" go test -json -tags "pam  bpf" -test.run=TestChaos ./lib/events/filesessions
    Updating git repository `[https://github.com/gravitational/boring`](https://github.com/gravitational/boring%60)
fatal: No names found, cannot describe anything.
GOWORK=off CGO_ENABLED=0 go -C build.assets/tooling run ./cmd/apiversion "18.6.4" > api/version.go
{"ImportPath":"github.com/gravitational/teleport/api","Action":"build-output","Output":"# github.com/gravitational/teleport/api\n"}
{"ImportPath":"github.com/gravitational/teleport/api","Action":"build-output","Output":"api/version.go:1:1: syntax error: package statement must be first\n"}
{"ImportPath":"github.com/gravitational/teleport/api","Action":"build-fail"}
make[1]: Leaving directory '/__w/teleport/teleport'
```

--- a/Makefile
+++ b/Makefile
@@ -946,7 +946,7 @@ test-helm-update-snapshots: helmunit/installed
 # Runs all Go tests except integration, called by CI/CD.
 #
 .PHONY: test-go
-test-go: test-go-prepare test-go-unit test-go-touch-id test-go-vnet-daemon test-go-tsh test-go-chaos
+test-go: test-go-unit test-go-touch-id test-go-vnet-daemon test-go-tsh test-go-chaos
 
 #
 # Runs a test to ensure no environment variable leak into build binaries.
@@ -980,7 +980,7 @@ test-go-prepare: ensure-webassets rdpclient $(TEST_LOG_DIR) $(VERSRC)
 test-go-unit: rdpclient
 test-go-unit: FLAGS ?= -race -shuffle on
 test-go-unit: SUBJECT ?= $(shell go list ./... | grep -vE 'teleport/(e2e|integration|tool/tsh|integrations/operator|integrations/access|integrations/lib)')
-test-go-unit:
+test-go-unit: test-go-prepare
 	$(CGOFLAG) go test -json -tags "$(PAM_TAG) $(RDPCLIENT_TAG) $(FIPS_TAG) $(BPF_TAG) $(LIBFIDO2_TEST_TAG) $(TOUCHID_TAG) $(PIV_TEST_TAG) $(VNETDAEMON_TAG) $(ADDTAGS)" $(PACKAGES) $(SUBJECT) $(FLAGS) $(ADDFLAGS) \
 		| go tool gotestsum --junitfile $(TEST_LOG_DIR)/unit-tests.xml --jsonfile $(TEST_LOG_DIR)/unit-tests.json --raw-command -- cat
 
@@ -995,7 +995,7 @@ test-go-unit-tbot:
 .PHONY: test-go-touch-id
 test-go-touch-id: FLAGS ?= -race -shuffle on
 test-go-touch-id: SUBJECT ?= ./lib/auth/touchid/...
-test-go-touch-id:
+test-go-touch-id: test-go-prepare
 ifneq ("$(TOUCHID_TAG)", "")
 	$(CGOFLAG) go test -json $(PACKAGES) $(SUBJECT) $(FLAGS) $(ADDFLAGS) \
 		| go tool gotestsum --junitfile $(TEST_LOG_DIR)/unit-tests-touchid.xml --jsonfile $(TEST_LOG_DIR)/unit-tests-touchid.json --raw-command -- cat
@@ -1033,7 +1033,7 @@ test-go-bench-root: $(BENCHFIND) | $(TEST_LOG_DIR)
 .PHONY: test-go-vnet-daemon
 test-go-vnet-daemon: FLAGS ?= -race -shuffle on
 test-go-vnet-daemon: SUBJECT ?= ./lib/vnet/daemon/...
-test-go-vnet-daemon:
+test-go-vnet-daemon: test-go-prepare
 ifneq ("$(VNETDAEMON_TAG)", "")
 	$(CGOFLAG) go test -json $(PACKAGES) $(SUBJECT) $(FLAGS) $(ADDFLAGS) \
 		| go tool gotestsum --junitfile $(TEST_LOG_DIR)/unit-tests-vnet.xml --jsonfile $(TEST_LOG_DIR)/unit-tests-vnet.json --raw-command -- cat
@@ -1043,14 +1043,14 @@ endif
 .PHONY: test-go-tsh
 test-go-tsh: FLAGS ?= -race -shuffle on
 test-go-tsh: SUBJECT ?= github.com/gravitational/teleport/tool/tsh/...
-test-go-tsh:
+test-go-tsh: test-go-prepare
 	$(CGOFLAG_TSH) go test -json -tags "$(PAM_TAG) $(FIPS_TAG) $(LIBFIDO2_TEST_TAG) $(TOUCHID_TAG) $(PIV_TEST_TAG) $(VNETDAEMON_TAG)" $(PACKAGES) $(SUBJECT) $(FLAGS) $(ADDFLAGS) \
 		| go tool gotestsum --junitfile $(TEST_LOG_DIR)/unit-tests-tsh.xml --jsonfile $(TEST_LOG_DIR)/unit-tests-tsh.json --raw-command -- cat
 
 # Chaos tests have high concurrency, run without race detector and have TestChaos prefix.
 .PHONY: test-go-chaos
 test-go-chaos: CHAOS_FOLDERS = $(shell find . -type f -name '*chaos*.go' | xargs dirname | uniq)
-test-go-chaos:
+test-go-chaos: test-go-prepare
 	$(CGOFLAG) go test -json -tags "$(PAM_TAG) $(FIPS_TAG) $(BPF_TAG)" -test.run=TestChaos $(CHAOS_FOLDERS) \
 		| go tool gotestsum --junitfile $(TEST_LOG_DIR)/unit-tests-chaos.xml --jsonfile $(TEST_LOG_DIR)/unit-tests-chaos.json --raw-command -- cat
 

